google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -236,6 +238,11 @@ private void reloadAppYamls(Project project) {
   protected void resetEditorFrom(@NotNull AppEngineDeploymentConfiguration configuration) {
     commonConfig.resetEditorFrom(configuration);
 
+    // if cloud project was not set at all, use active cloud project.
+    if (commonConfig.getProjectSelector().getSelectedProject() == null) {
```

Comment:
I was kinda thinking that instead of having each consumer check for thie new 'default' project this, that this could be handled internally by the project selector.

i.e. if the project persisted in the specific config by the user is null (or not set), then the project selector itself will be responsible for showing its default project (if one exists).

do you think this is doable / makes sense?

Reply:
ditto.

Reply:
I think I can just refactor this code to be called less explicitly, i.e. when settings editor is ready, always load active project first, not in `resetEditor` checking for null.

If we load active project automatically in setSelectedProject(null), we must be sure everything is ready - UI, state, listeners so active project will be properly loaded and all listeners triggered. This is a very strong side effect vs. just doing it explicitly when one as owner of UI knows it is ready. WDYT?